### ISSUE TITLE
Horizontal Recoil (plus probability of it)

### ISSUE THREAD

**Issue by Kirby859** [FlansMods/FlansMod#139]
YEAHHH 100th open issue so far

Right now, most of my guns (heh, I remember that pulse rifle I made that recoiled down) recoil straight up. It's a little too easy to control, but if I set the recoil really high, the gun's recoil seems over-exaggerated. If I set it low, then the recoil is boring and easy to control. Groupings seem a lot more randomized and harder to control with horizontal recoil:

![2013-05-26_00001](https://f.cloud.github.com/assets/3915213/565640/3175d17c-c629-11e2-84a6-3bfe374dcbd6.jpg)
Left:  MP5A4 from Firearms: Source (a little more recoil)
Right:  CF-05 from Firearms:  Source (a little less recoil)

 I suggest an adjustable amount of recoil:

---

A.
1 field for horizontal recoil amount. This might be SidewaysRecoil or HorizontalRecoil. I might set it to SidewaysRecoil 2 (or HorizontalRecoil 2) so the gun would recoil a little bit to the side (with a 50/50 chance for left or right when it does recoil) occasionally. How often? Well:

There should also be a field called SidewaysRecoilProbablility (or something shorter, preferably). This could be set to a value between 0 and 100\* (or whatever Flan decides to code it as), which would determine the probability that the gun would recoil sideways.

So with all of this, my recoil settings might be:

//other stuff
Recoil 3
SidewaysRecoil 2
SidewaysRecoilProbability 40
//other stuff

SidewaysRecoilProbability might be divided by 2 to have LeftRecoilProbability be 20% and RightRecoilProbability be 20%, with NoSidewaysRecoilProbability be the remainder (which is 60%).

This seems to be the most simple form. A more customizable version might be:

---

B.
2 fields for horizontal recoil amount. I suggest LeftRecoil and RightRecoil. LeftRecoil would be the amount that the gun recoils left (when the mod decides that the gun should recoil left). RightRecoil would be the amount that the gun recoils right (when the mod decides that the gun should recoil right). How often should the gun recoil left? How about right? Well:

There should be 2 or 3 fields for the probability of the gun recoiling left or right. There are 2 ways (that I can think of right now) this could be implemented.

1.
LeftRecoilProbability -- This field, which would probably be a number between 0 and 100\* (for percentage), would determine how often the gun should recoil left. If I set it to 40, then the gun should recoil ABOUT (hopefully randomly) every 2 out of 5 shots fired. Along with that, there would be:
RightRecoilProbability -- This field would be the same as LeftRecoilProbability, except for recoiling right.

If Flan is able to have the mod calculate the remainder of 100-LeftRecoilProbability+RightRecoilProbability, then he should be able to have that be the probability for neither left or right recoil (just up) (NoSidewaysRecoilProbability).

Upwards recoil should always be 100 percent. Probably.

2.
Number 1 should be implemented, except for the NoSidewaysRecoilProbability calculation. If Flan is somehow unable to have the mod automatically calculate that, then I guess that he should be able to make a NoSidewaysRecoilProbability (hopefully shorter) field. This would be the chance (0 to 100*) that would make the gun just recoil straight up.

So with all of this, here are the ways that the feature would be used by pack makers:

1.
//other stuff
LeftRecoil 2
LeftRecoilProbability 40
RightRecoil 3
RightRecoilProbability 20
//other stuff
//NoSidewaysRecoilProbability would be calculated by 100-(40+20)=40

2.
LeftRecoil 4
LeftRecoilProbability 10
RightRecoil 2
RightRecoilProbability 30
NoSidewaysRecoilProbability 60
//LeftRecoilProbability + RightRecoilProbability + NoSidewaysRecoilProbability HAS TO EQUAL 100 (or some other number (10?) that Flan uses.

---

This was probably the longest suggestion that I've written up.

tl;dr Sideways recoil, either have just:
2 fields (SidewaysRecoil, SidewaysRecoilProbability)

4 fields (LeftRecoil, LeftRecoilProbability, RightRecoil, RightRecoilProbability)

or 5 fields (LeftRecoil, LeftRecoilProbability, RightRecoil, RightRecoilProbability, NoSidewaysRecoilProbability)

---

*0-100, or 0-10 for smaller simpler numbers, or some other arbitrary range of numbers.
Also, shorter field names would be amazing. (seriously, NoSidewaysRecoilProbability?)

---

Hi, godgodgodgo. If I forgot to write something important, please tell me. I just kind of vomit out these suggestions after I think about them.
